tensorflow/python/distribute/mirrored_strategy.py

```diff
@@ -503,8 +481,15 @@ def _experimental_make_numpy_dataset(self, numpy_input, session):
 
   def _experimental_distribute_datasets_from_function(self, dataset_fn,
                                                       options):
+    if options.replication_mode == distribute_lib.InputReplicationMode.PER_REPLICA:
+      self._input_workers_devices = (
```

Comment:
do you want to set this to `self._input_workers_devices` ? I thought we just want to pass this on to `self._input_workers_with_options(` but not set it on the object.

Reply:
Unfortunately I have to rebuild the _input_workers_devices because the tuples are fundamentally different in the two cases.